crates/services/upgradable-executor/src/storage_access_recorder.rs

```diff
@@ -0,0 +1,48 @@
+use fuel_core_storage::{
+    kv_store::{
+        KeyValueInspect,
+        StorageColumn,
+        Value,
+    },
+    Result as StorageResult,
+};
+use fuel_core_types::services::executor::StorageReadReplayEvent;
+use parking_lot::Mutex;
+use std::sync::Arc;
+
+pub struct StorageAccessRecorder<S>
+where
+    S: KeyValueInspect,
+{
+    pub storage: S,
+    pub record: Arc<Mutex<Vec<StorageReadReplayEvent>>>,
```

Comment:
Instead of `Arc<Mutex<_>>` I think you could use `RefCell`. I will make `StorageAccessRecorder` non `Send`, but I don't think that we have cases where you need it to be `Send`. But maybe I'm not right

Reply:
We require it to be `Send` here: https://github.com/FuelLabs/fuel-core/blob/686f8ce24ec8e1e7298dd9b3c671ffe6adbdebc8/crates/services/upgradable-executor/src/instance.rs#L304